src/main/routes/static-file-route.injectable.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type {
+  LensApiRequest, Route,
+} from "../router";
+import logger from "../logger";
+import { routeInjectionToken } from "../router/router.injectable";
+import {
+  appName,
+  isDevelopment,
+  publicPath,
+  webpackDevServerPort,
+} from "../../common/vars";
+import path from "path";
+import readFileInjectable from "../../common/fs/read-file.injectable";
+import type { SupportedFileExtension } from "../router-content-types";
+import { contentTypes } from "../router-content-types";
+
+interface Dependencies {
+  readFile: (path: string) => Promise<Buffer>
+}
+
+const staticFileRoute = ({ readFile }: Dependencies) => async ({
+  params,
+  raw: { req },
+}: LensApiRequest) => {
+  const staticPath = path.resolve(__static);
+
+
+  let filePath = params.path;
+
+  for (let retryCount = 0; retryCount < 5; retryCount += 1) {
+    const asset = path.join(staticPath, filePath);
+
+    const normalizedFilePath = path.resolve(asset);
+
+    if (!normalizedFilePath.startsWith(staticPath)) {
+      return { statusCode: 404 };
+    }
+
+    try {
+      const filename = path.basename(req.url);
+
+      // redirect requests to [appName].js, [appName].html /sockjs-node/ to webpack-dev-server (for hot-reload support)
+      const toWebpackDevServer =
+        filename.includes(appName) ||
+        filename.includes("hot-update") ||
+        req.url.includes("sockjs-node");
+
```

Comment:
Lines related to `webpack-dev-server` where removed from `router.ts` in https://github.com/lensapp/lens/pull/4894 and here probably should be too.

Reply:
Let's figure this out when we start merging 5.5.0 stuff to master. Good catch.

Reply:
Merge conflict resolved.